src/vaults/AsyncRequests.sol

```diff
@@ -230,7 +236,7 @@ contract AsyncRequests is BaseInvestmentManager, IAsyncRequests {
 
         // Mint to escrow. Recipient can claim by calling deposit / mint
         IShareToken shareToken = IShareToken(IAsyncVault(vault_).share());
-        shareToken.mint(address(escrow), shares);
+        shareToken.mint(poolEscrowProvider.escrow(poolId), shares);
```

Comment:
QUESTION

Won't this call into the `hook` too and then block as it is not he `ESCROW_HOOK_ID`?